pkg/kubelet/dockershim/docker_image.go

```diff
@@ -101,8 +101,6 @@ func (ds *dockerService) RemoveImage(image *runtimeapi.ImageSpec) error {
 			}
 		}
 		return nil
-	} else if err != nil && libdocker.IsImageNotFoundError(err) {
```

Comment:
It should return err if error is not `ImageNotFoundError` and not nil.

Also should add a comment for the reason:  dockerclient.InspectImageByID doesn't work with digest and repoTags, it is safe to continue removing it since there is another check below.